File: go/leetCode/771_Jewels_and_Stones/solution.js
/*
 * @Author: wcy
 * @Date:   2018-07-21 21:15:25
 * @Last Modified by:   wcy
 * @Last Modified time: 2018-07-21 22:28:56
 */

let numJewelsInStones_1 = function(J, S) {
	let result = 0;
	for (let j of J) {
		for (let s of S) {
			if (j !== s) {
				continue;
			}
			result++;
		}
	}
	return result;
}

let numJewelsInStones_2 = function(J, S) {
	let result = 0;
	for (let s of S) {
		if (J.indexOf(s) < 0) {
			continue;
		}
		result++;
	}
	return result;
}

let J = 'Aads';
let S = 'aAAsdfsgfsg';

console.log(numJewelsInStones_3(J, S))

File: go/leetCode/771_Jewels_and_Stones/solution.go
/*
* @Author: wcy
* @Date:   2018-07-21 21:56:50
* @Last Modified by:   wcy
* @Last Modified time: 2018-07-21 22:11:42
 */

package main

import (
	"fmt"
	"strings"
)

func numJewelsInStones_1(J string, S string) int {
	result := 0
	for _, v := range S {
		if strings.Index(J, string(v)) < 0 {
			continue
		}
		result++
	}
	return result
}

func numJewelsInStones_2(J string, S string) int {
	result := 0
	for _, v := range J {
		result += strings.Count(S, string(v))
	}
	return result
}

func main() {
	var J string = "aA"
	var S string = "afaaf"
	fmt.Println(numJewelsInStones_1(J, S))
	fmt.Println(numJewelsInStones_2(J, S))
}
